internal/jobworker/grpc/server.go

```diff
@@ -0,0 +1,230 @@
+	); err != nil {
+		logger.Errorf("starting Job; error: %v", err)
+		return nil, status.Error(codes.Internal, "error starting job")
+	}
+
+	logger.Infof("Job started; ID: %v", j.ID)
+	return &pb.StartResponse{
+		JobId:   j.ID.String(),
+		Command: req.Command,
+		Status: &pb.StatusDetail{
+			Status:   toStatus(j.Status()),
+			ExitCode: int32(j.ExitCode()),
+		},
+		Limits: req.Limits,
+	}, nil
+}
+
+func (jw JobWorker) Stop(ctx context.Context, req *pb.StopRequest) (*pb.StopResponse, error) {
+	user, ok := jw.userSvc.User(ctx)
+	if !ok {
+		return nil, status.Error(codes.Unauthenticated, "unauthenticated")
+	}
+
+	if req.JobId == "" {
+		return nil, status.Error(codes.InvalidArgument, validator.Format("empty job ID"))
+	}
+
+	j, err := jw.fetchJob(ctx, user, req.JobId)
+	if err != nil {
+		return nil, err
+	}
+
+	if j.Status() != job.Running {
+		return nil, status.Error(codes.FailedPrecondition, "job is not running")
+	}
+
+	if err := jw.jobSvc.StopJob(ctx, j.ID); err != nil {
+		logger.Errorf("stop job; job: %s, error: %v", j.ID, err)
+		return nil, status.Error(codes.Internal, "error stopping job")
+	}
+
+	return &pb.StopResponse{}, nil
+}
+
+func (jw JobWorker) Status(ctx context.Context, req *pb.StatusRequest) (*pb.StatusResponse, error) {
+	user, ok := jw.userSvc.User(ctx)
+	if !ok {
+		return nil, status.Error(codes.Unauthenticated, "unauthenticated")
+	}
+
+	if req.JobId == "" {
+		return nil, status.Error(codes.InvalidArgument, validator.Format("empty job ID"))
+	}
+
+	j, err := jw.fetchJob(ctx, user, req.JobId)
+	if err != nil {
+		return nil, err
+	}
+
+	return &pb.StatusResponse{
+		Status: &pb.StatusDetail{
+			Status:   toStatus(j.Status()),
+			ExitCode: int32(j.ExitCode()),
+		},
+	}, nil
+}
+
+func (jw JobWorker) Output(req *pb.OutputRequest, stream pb.JobWorkerService_OutputServer) error {
+	user, ok := jw.userSvc.User(stream.Context())
+	if !ok {
+		return status.Error(codes.Unauthenticated, "unauthenticated")
+	}
+
+	if req.JobId == "" {
+		return status.Error(codes.InvalidArgument, validator.Format("empty job ID"))
+	}
+
+	j, err := jw.fetchJob(stream.Context(), user, req.JobId)
+	if err != nil {
+		return err
+	}
+
+	ctx, cancel := context.WithCancel(stream.Context())
+	defer cancel()
+
+	outputc := make(chan []byte, streamBuffer)
+	go func() {
+		if err := j.StreamOutput(ctx, outputc, chunkSize); err != nil {
+			logger.Errorf("streaming output from job; job: %s, error: %v", j.ID, err)
+		}
+		close(outputc)
+	}()
+
+	for b := range outputc {
+		if err := stream.Send(&pb.OutputResponse{Output: b}); err != nil {
+			logger.Errorf("streaming output to client; job: %s, error: %s", j.ID, err)
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (jw JobWorker) fetchJob(ctx context.Context, user string, jobID string) (*job.Job, error) {
+	id, err := uuid.Parse(jobID)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, validator.Format("job ID not UUID"))
+	}
+
+	j, err := jw.jobSvc.FetchJob(ctx, id)
```

Comment:
My 2c: Make FetchJob (and all other funcs) take the ID as a string. The underlying format doesn't matter for API users, plus we get to ditch the uuid.Parse above.